test/code/scxcorelib/pal/scxfile_test.cpp

```diff
@@ -631,6 +634,47 @@ class SCXFileTest : public CPPUNIT_NS::TestFixture {
         }
         SCXFile::Delete(path);
     }
+    
+    void TestCreateTempFileValidDirectory()
+    {
+    	wstring content(L"This is the file content\non the temp file\n");
+    	SCXFilePath path = SCXFile::CreateTempFile(content,L"/tmp/");
+    	CPPUNIT_ASSERT(SCXFile::Exists(path));
+    	SCXCoreLib::SCXHandle<std::wfstream> fileStream = SCXFile::OpenWFstream(path, std::ios::in);
+    	std::wostringstream compare;
+    	(*fileStream) >> compare.rdbuf();
+    	if (content != compare.str()) 
+    	{
+    	    wcout << L"content: " << content << std::endl;
+    	    wcout << L"compare: " << compare.str() << std::endl;
+    	    CPPUNIT_ASSERT(content == compare.str());
+    	}
+    	SCXFile::Delete(path);
+    }
+    
+    void TestCreateTempFileInvalidDirectory()
+    {
+        wstring content(L"This is the file content\non the temp file\n");
+        SCXUNIT_ASSERT_THROWN_EXCEPTION(SCXFile::CreateTempFile(content,L"/not/valid/directory/"), SCXFilePathNotFoundException, L"/not/valid/directory");
+    }
+    
+    void TestCreateTempFileNonDefaultDirectory()
+    {
+    	wstring content(L"This is the file content\non the temp file\n");
+    	// Creating Temp file in CWD
+    	SCXFilePath path = SCXFile::CreateTempFile(content,L"./");
+    	CPPUNIT_ASSERT(SCXFile::Exists(path));
+    	SCXCoreLib::SCXHandle<std::wfstream> fileStream = SCXFile::OpenWFstream(path, std::ios::in);
+    	std::wostringstream compare;
+    	(*fileStream) >> compare.rdbuf();
+    	if (content != compare.str())
+    	{
+    		wcout << L"content: " << content << std::endl;
+    		wcout << L"compare: " << compare.str() << std::endl;
+    		CPPUNIT_ASSERT(content == compare.str());
+    	}
+    	SCXFile::Delete(path);
```

Comment:
Close, but not quite. The problem with this: If there's an assertion, then the temp file is left around. We prefer using SCXCoreLib::SelfDeletingFilePath for that, as that cleans up the file "no matter what". For an example, please see (in the PAL): test/code/scxsystemlib/disk/staticlogicaldiskpal_test.cpp. This macro is defined in test/code/include/testutils/scxtestutils.h.